src/components/input/input.js

```diff
@@ -404,6 +404,9 @@ function mdMaxlengthDirective($animate) {
       if (!angular.isNumber(maxlength) || maxlength < 0) {
         return true;
       }
+      if (angular.isNumber(modelValue)) {
+        modelValue = modelValue.toString();
```

Comment:
This will fail for very large numbers:

```js
var num = 2000000929340720349810293480192834091823409172394712934;
num.toString() // => "2.0000009293407202e+54"
num.length // => 22; should be 55
```

Instead of checking the model value, perhaps we should check the element value in this case? Does that provide what we need?